src/Service.Tests/SqlTests/RestApiTests/RestBodyNonStrictModeTests/RestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,315 @@
+            // in the underlying table are included in the request body for PUT operation,
+            // they are just ignored when we do not operate in strict mode for REST.
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""non_existing_field"": ""5""
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithExtraneousFieldsInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
+
+            // Validate that when a computed field (here 'last_sold_on_date') is included the request body for PUT update operation,
+            // it is just ignored when we do not operate in strict mode for REST. Successful execution of the PUT request confirms
+            // that we did not attempt to update the value of 'last_sold_on_update' field.
+            requestBody = @"
+            {
+                ""book_name"": ""New book"",
+                ""copies_sold"": 101,
+                ""last_sold_on"": ""2023-09-12 05:30:30""
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/1",
+                    queryString: null,
+                    entityNameOrPath: _entityWithReadOnlyFields,
+                    sqlQuery: GetQuery("PutOneUpdateWithComputedFieldInRequestBody"),
+                    operationType: EntityActionOperation.Upsert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK
+                );
+
+            // Validate that when a computed field (here 'last_sold_on_date') is included the request body for PUT insert operation,
+            // it is just ignored when we do not operate in strict mode for REST. Successful execution of the PUT request confirms
+            // that we did not attempt to insert a value for 'last_sold_on_update' field.
+            requestBody = @"
+            {
+                ""book_name"": ""New book"",
+                ""copies_sold"": 101
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "id/2",
+                    queryString: null,
+                    entityNameOrPath: _entityWithReadOnlyFields,
+                    sqlQuery: GetQuery("PutOneInsertWithComputedFieldInRequestBody"),
+                    operationType: EntityActionOperation.Upsert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.Created,
+                    expectedLocationHeader: "id/2"
+                );
```

Comment:
Comment outdated and computed field mentioned is not in the request body. What is this testing?

Reply:
Missed including the field in the body, doing it now.